packages/@internationalized/date/src/conversion.ts

```diff
@@ -0,0 +1,203 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+// Portions of the code in this file are based on code from the TC39 Temporal proposal.
+// Original licensing can be found in the NOTICE file in the root directory of this source tree.
+
+import {Calendar} from './types';
+import {CalendarDate, CalendarDateTime, Time} from './CalendarDate';
+import {GregorianCalendar} from './calendars/GregorianCalendar';
+
+function epochFromDate(date: CalendarDateTime) {
+  date = toCalendar(date, new GregorianCalendar());
+  return epochFromParts(date.year, date.month, date.day, date.hour, date.minute, date.second, date.millisecond);
+}
+
+function epochFromParts(year: number, month: number, day: number, hour: number, minute: number, second: number, millisecond: number) {
+  // Note: Date.UTC() interprets one and two-digit years as being in the
+  // 20th century, so don't use it
+  let date = new Date();
+  date.setUTCHours(hour, minute, second, millisecond);
+  date.setUTCFullYear(year, month - 1, day);
+  return date.getTime();
+}
+
+export function getTimeZoneOffset(ms: number, timeZone: string) {
+  let {year, month, day, hour, minute, second} = getTimeZoneParts(ms, timeZone);
+  let utc = epochFromParts(year, month, day, hour, minute, second, 0);
+  return utc - ms;
+}
+
+const formattersByTimeZone = new Map<string, Intl.DateTimeFormat>();
+
+function getTimeZoneParts(ms: number, timeZone: string) {
+  let formatter = formattersByTimeZone.get(timeZone);
+  if (!formatter) {
+    formatter = new Intl.DateTimeFormat('en-US', {
+      timeZone,
+      hour12: false,
+      era: 'short',
+      year: 'numeric',
+      month: 'numeric',
+      day: 'numeric',
+      hour: 'numeric',
+      minute: 'numeric',
+      second: 'numeric'
+    });
+
+    formattersByTimeZone.set(timeZone, formatter);
+  }
+
+  let parts = formatter.formatToParts(new Date(ms));
+  let namedParts: {[name: string]: string} = {};
+  for (let part of parts) {
+    if (part.type !== 'literal') {
+      namedParts[part.type] = part.value;
+    }
+  }
+
+  return {
+    year: namedParts.era === 'BC' ? -namedParts.year + 1 : +namedParts.year,
+    month: +namedParts.month,
+    day: +namedParts.day,
+    hour: namedParts.hour === '24' ? 0 : +namedParts.hour, // bugs.chromium.org/p/chromium/issues/detail?id=1045791
```

Comment:
In the external bugs list?

Reply:
I think it came from one of the libraries I ported but I will add it